api_end_end_testing/src/test/java/tds/student/api/tests/StudentApiTests.java

```diff
@@ -0,0 +1,726 @@
+            .body(studentInfo)
+        .when()
+            .post(uriLocation + stateAbbrev)
+        .then()
+            .statusCode(400);
+    }
+
+    @Test
+    public void shouldNotCreateUpdateStudentWithInvalidGradeLevel() {
+        String ssid = createRandomStudentSsid();
+        String stateAbbrev = "CA";
+
+        // Create a student
+        StudentInfo studentInfo = createStudent(ssid, stateAbbrev);
+
+        // Test and invalid state abbreviation
+        studentInfo.setGradeLevelWhenAssessed("123456");
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(studentInfo)
+        .when()
+            .post(uriLocation + stateAbbrev)
+        .then()
+            .statusCode(400)
+            .body("messages.gradeLevelWhenAssessed[0]", startsWith("Grade level when assessed should be"));
+
+    }
+
+    @Test
+    public void shouldNotCreateUpdateStudentWithInvalidLanguageCode() {
+        String ssid = createRandomStudentSsid();
+        String stateAbbrev = "CA";
+
+        // Create a student
+        StudentInfo studentInfo = createStudent(ssid, stateAbbrev);
+
+        // Test and invalid state abbreviation
+        studentInfo.setLanguageCode("ENGL");
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(studentInfo)
+        .when()
+            .post(uriLocation + stateAbbrev)
+        .then()
+            .statusCode(400)
+            .body("messages.languageCode[0]", startsWith("Language Codes are invalid"));
+    }
+
+    @Test
+    public void shouldNotCreateUpdateStudentWithInvalidFirstEntryDate() {
+        String ssid = createRandomStudentSsid();
+        String stateAbbrev = "CA";
+
+        // Create a student
+        StudentInfo studentInfo = createStudent(ssid, stateAbbrev);
+
+        // Test and invalid state abbreviation
+        studentInfo.setFirstEntryDateIntoUsSchool("1930-01-01");
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(studentInfo)
+        .when()
+            .post(uriLocation + stateAbbrev)
+        .then()
+            .statusCode(400)
+            .body("messages.firstEntryDateIntoUsSchool[0]", startsWith("First Entry Date Into US School should always be after the year of birth date"));
+    }
+
+    @Test
+    public void shouldNotCreateUpdateStudentWithInvalidLepEntryDate() {
+        String ssid = createRandomStudentSsid();
+        String stateAbbrev = "CA";
+
+        // Create a student
+        StudentInfo studentInfo = createStudent(ssid, stateAbbrev);
+
+        // Test and invalid state abbreviation
+        studentInfo.setLepEntryDate("214-1999-12-01");
+
+        given()
+            .contentType(ContentType.JSON)
+            .header(authHeader)
+            .body(studentInfo)
+        .when()
+            .post(uriLocation + stateAbbrev)
+        .then()
+            .statusCode(400);
```

Comment:
Does this not return a body?  We verify on the tests above but this one and the one below does not.